### ISSUE TITLE
[1.3.0-RC4] no caching in crossbuild

### ISSUE THREAD

**Issue by OlegYch** [sbt/sbt#4986]
## steps
sbt version: 1.3.0-RC4
`build.sbt`
```
lazy val a = project.in(file(".")).settings(
  crossScalaVersions := List("2.12.9", "2.13.0", "2.11.12"),
)
```
`src/main/scala/a.scala`
```
object A {
  def a = 1
}
```

## problem
running `+compile` recompiles the file even if it didn't change
`compile` works fine

## expectation
`+compile` caches compilation results

## notes
works fine with RC3

**Comment by eatkins** [comment]
Thanks for the report! Should be fixed by #4987.

**Comment by eed3si9n** [comment]
Merged #4987
